### PR TITLE
Do not call "bash -c" unnecessarily as entry points

### PR DESCRIPTION
**Pull request by ralonsoh**
Closes: [OSPRH-680](https://issues.redhat.com//browse/OSPRH-680)

### CODE REVIEW

File: pkg/ovncontroller/daemonset.go [hunk 1]
```diff
@@ -131,14 +131,11 @@ func DaemonSet(
 	}
 
 	ovnControllerCmd = []string{
-		"/bin/bash", "-c",
+		"/usr/local/bin/container-scripts/net_setup.sh && ovn-controller",
```
Comment (booxter) [inline pkg/ovncontroller/daemonset.go:134]: I don't think this will work, because && is handled by shell.

File: pkg/ovncontroller/daemonset.go [hunk 2]
```diff
@@ -131,14 +131,11 @@ func DaemonSet(
...
 	}
 	ovnControllerArgs = []string{
 		strings.Join(
-			append(
```
Comment (booxter) [inline pkg/ovncontroller/daemonset.go:138]: looks like this doesn't compile, as per CI

**Review by booxter — CHANGES_REQUESTED** [review]
Doesn't compile, and I think you have to have a shell to do &&. Either we find a different way to do setup (with job / preStart hook), or have a script that would start ovn-controller for us; or do nothing.

**Comment by ralonsoh** [comment]
@booxter @karelyatin @slawqo I think this patch is ready for review, thanks in advance.

**Comment by karelyatin** [comment]
@ralonsoh LGTM for the changes done here, but the linked but also mentions about bash usage in ovncontroller but i don't see any changes/comment for that either in PR or in Jira and also the comment says Closes Jira.

**Comment by ralonsoh** [comment]
> @ralonsoh LGTM for the changes done here, but the linked but also mentions about bash usage in ovncontroller but i don't see any changes/comment for that either in PR or in Jira and also the comment says Closes Jira.

@karelyatin when the ovn-controller command [1] is executed outside the bash call, it always returns an error with the socket file. Also when dumb-init is used. The directory /run/openvswitch exists. E.g.:
```
ospng@r7rhoso:~/ovn-operator$ oc logs pods/$p -n kuttl3                                                                                                                                                                                                     
[dumb-init] ovn-controller --pidfile unix:/run/openvswitch/db.sock: No such file or directory
```
[1]https://github.com/openstack-k8s-operators/ovn-operator/blob/9bdb826b7218c51a029b8c213850022a70cea4fd/pkg/ovncontroller/daemonset.go#L39C4-L39C18

**Comment by karelyatin** [comment]
> > @ralonsoh LGTM for the changes done here, but the linked but also mentions about bash usage in ovncontroller but i don't see any changes/comment for that either in PR or in Jira and also the comment says Closes Jira.
> 
> @karelyatin when the ovn-controller command [1] is executed outside the bash call, it always returns an error with the socket file. Also when dumb-init is used. The directory /run/openvswitch exists. E.g.:
> 
> ```
> ospng@r7rhoso:~/ovn-operator$ oc logs pods/$p -n kuttl3                                                                                                                                                                                                     
> [dumb-init] ovn-controller --pidfile unix:/run/openvswitch/db.sock: No such file or directory
> ```
> 
> [1]https://github.com/openstack-k8s-operators/ovn-operator/blob/9bdb826b7218c51a029b8c213850022a70cea4fd/pkg/ovncontroller/daemonset.go#L39C4-L39C18

@ralonsoh but dumb-init shouldn't be needed just for this, i did tried locally removing bash and seems to work.

diff --git a/pkg/ovncontroller/daemonset.go b/pkg/ovncontroller/daemonset.go
index 09ce8e3..0b5491d 100644
--- a/pkg/ovncontroller/daemonset.go
+++ b/pkg/ovncontroller/daemonset.go
@@ -14,7 +14,6 @@ package ovncontroller
 
 import (
        "fmt"
-       "strings"
 
        "github.com/openstack-k8s-operators/lib-common/modules/common/env"
        "github.com/openstack-k8s-operators/lib-common/modules/common/tls"
@@ -35,8 +34,8 @@ func CreateOVNDaemonSet(
        volumes := GetOVNControllerVolumes(instance.Name, instance.Namespace)
        mounts := GetOVNControllerVolumeMounts()
 
-       args := []string{
-               "ovn-controller --pidfile unix:/run/openvswitch/db.sock",
+       cmd := []string{
+               "ovn-controller", "--pidfile", "unix:/run/openvswitch/db.sock",
        }
 
        // add OVN dbs cert and CA
@@ -56,7 +55,7 @@ func CreateOVNDaemonSet(
                        mounts = append(mounts, instance.Spec.TLS.CreateVolumeMounts(nil)...)
                }
 
-               args = append(args, []string{
+               cmd = append(cmd, []string{
                        fmt.Sprintf("--certificate=%s", ovn_common.OVNDbCertPath),
                        fmt.Sprintf("--private-key=%s", ovn_common.OVNDbKeyPath),
                        fmt.Sprintf("--ca-cert=%s", ovn_common.OVNDbCaCertPath),
@@ -72,8 +71,8 @@ func CreateOVNDaemonSet(
        containers := []corev1.Container{
                {
                        Name:    "ovn-controller",
-                       Command: []string{"/bin/bash", "-c"},
-                       Args:    []string{strings.Join(args, " ")},
+                       Command: cmd,
+                       //Args:    []string{strings.Join(cmd, " ")},
                        Lifecycle: &corev1.Lifecycle{
                                PreStop: &corev1.LifecycleHandler{
                                        Exec: &corev1.ExecAction{

**Comment by ralonsoh** [comment]
> @ralonsoh but dumb-init shouldn't be needed just for this, i did tried locally removing bash and seems to work. https://paste.opendev.org/show/bcDn2u4DRCaMbNqnJvfk/

I did something similar some PS ago [1] and it wasn't working. The problem was to divide the command string... Thanks for your help.

[1]https://github.com/openstack-k8s-operators/ovn-operator/commit/40f649e0d2ea6070e1901301cfc9b168fcf5c2da#diff-853730e1a0e56623824bb603929544f54d838010a75b27a62b285ae738064744

**Comment by ralonsoh** [comment]
Ok, I think now is ready (again).

**Review by booxter — APPROVED** [review]

**Comment by booxter** [comment]
Not sure why it didn't create `lgtm` label before while Yatin put `/lgtm` in the comment. But it's all good now, should merge now.

**Comment by karelyatin** [comment]
> Not sure why it didn't create `lgtm` label before while Yatin put `/lgtm` in the comment. But it's all good now, should merge now.

I recall seeing something similar in past where bot didn't add the label. Readding helped, but don't know about root cause. thx for noticing it.

**Comment by karelyatin** [comment]
> ```
>          --- FAIL: kuttl/harness/ovn_db_delete (234.92s) 
> ```
> 
> It may be worth understanding why this failed before merging. It may be not related to this patch, but if so, it would be nice to understand what other issue triggers it, and to make sure there's a Jira issue for this.

Yes it's unrelated, As seen in [1] pod-0 got deleted first, pod-1/pod-2 stuck Terminating as waiting for cluster leave. We have a ticket for this [2] and that have a related FDP issue

[1] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openstack-k8s-operators_ovn-operator/249/pull-ci-openstack-k8s-operators-ovn-operator-main-ovn-operator-build-deploy-kuttl/1839362702133170176/artifacts/ovn-operator-build-deploy-kuttl/openstack-k8s-operators-gather/artifacts/must-gather/quay-io-openstack-k8s-operators-openstack-must-gather-sha256-3d1f620ae3617fbd6371f55aa948b34e9daf31a461d3c557cf1a7c2db38d0385/namespaces/ovn-kuttl-tests/all_resources.log
[2] https://issues.redhat.com/browse/OSPRH-7626

**Comment by karelyatin** [comment]
> > ```
> >          --- FAIL: kuttl/harness/ovn_db_delete (234.92s) 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > It may be worth understanding why this failed before merging. It may be not related to this patch, but if so, it would be nice to understand what other issue triggers it, and to make sure there's a Jira issue for this.
> 
> Yes it's unrelated, As seen in [1] pod-0 got deleted first, pod-1/pod-2 stuck Terminating as waiting for cluster leave. We have a ticket for this [2] and that have a related FDP issue
> 
> [1] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openstack-k8s-operators_ovn-operator/249/pull-ci-openstack-k8s-operators-ovn-operator-main-ovn-operator-build-deploy-kuttl/1839362702133170176/artifacts/ovn-operator-build-deploy-kuttl/openstack-k8s-operators-gather/artifacts/must-gather/quay-io-openstack-k8s-operators-openstack-must-gather-sha256-3d1f620ae3617fbd6371f55aa948b34e9daf31a461d3c557cf1a7c2db38d0385/namespaces/ovn-kuttl-tests/all_resources.log [2] https://issues.redhat.com/browse/OSPRH-7626

pushed a workaround https://github.com/openstack-k8s-operators/ovn-operator/pull/356

**Pull request merged**